lib/vmpooler/api/v1.rb

```diff
@@ -66,8 +66,9 @@ def account_for_starting_vm(template, vm)
           backend.hget('vmpooler__token__' + request.env['HTTP_X_AUTH_TOKEN'], 'user')
         )
 
-        if config['vm_lifetime_auth'].to_i > 0
-          backend.hset('vmpooler__vm__' + vm, 'lifetime', config['vm_lifetime_auth'].to_i)
+        pool_lifetime_auth = pools.find {|p| p["name"] == template }['vm_lifetime_auth']
+        if (pool_lifetime_auth || config['vm_lifetime_auth']).to_i > 0
+          backend.hset('vmpooler__vm__' + vm, 'lifetime', (pool_lifetime_auth || config['vm_lifetime_auth']).to_i)
```

Comment:
:+1: on @mckern's suggestion here. There's a lot of copy-pasta inside vmpooler and I think we're well served by extracting methods with intention-revealing names when we can.